File: src/exercise1/python/counter.py
from threading import Thread

counter = 0
rounds = 100000


def run():
    for i in range(rounds):
        global counter
        counter += 1


if __name__ == '__main__':
    t1 = Thread(target=run)
    t2 = Thread(target=run)

    t1.start()
    t2.start()

    t1.join()
    t2.join()
    print(counter)

File: src/exercise1/go/counter.go
package main

import (
	"fmt"
	"sync"
)

var counter = 0

func run(wg *sync.WaitGroup, rounds int) {
	defer wg.Done()

	for i := 0; i < rounds; i++ {
		counter++
	}
}

func main() {
	var wg sync.WaitGroup

    wg.Add(2)
    go run(&wg, 100000)
    go run(&wg, 100000)

	wg.Wait()
    fmt.Println(counter)
}
